extractor/src/extract/extractor.py

```diff
@@ -199,67 +126,16 @@ def run_sql_query(self, query, resource_id=None):
 
         return query.yield_per(CHUNK_SIZE)
 
-    def batch_size(self, analysis) -> int:
-        logger.info(
-            f"Start computing batch size for resource {analysis.definition_id}",
-            extra={"resource_id": analysis.resource_id},
-        )
-        pk_column = self.get_column(analysis.primary_key_column)
-        base_query = self.session.query(func.count(distinct(pk_column)))
-        query_w_joins = self.apply_joins(base_query, analysis.joins)
-        query_w_filters = self.apply_filters(query_w_joins, analysis, None)
-        logger.info(
-            f"Sql query to compute batch size: {query_w_filters.statement}",
-            extra={"resource_id": analysis.resource_id},
-        )
-        res = query_w_filters.session.execute(query_w_filters)
-
-        return res.scalar()
-
-    def get_columns(self, columns: List[SqlColumn]) -> List[AlchemyColumn]:
-        """ Get the sql alchemy columns corresponding to the SqlColumns (custom type)
-        from the analysis.
-        """
-        return [self.get_column(col) for col in columns]
-
-    def get_column(self, column: SqlColumn) -> AlchemyColumn:
-        """ Get the sql alchemy column corresponding to the SqlColumn (custom type)
-        from the analysis.
-        """
-        table = self.get_table(column)
-        # Note that we label the column manually to avoid collisions and
-        # sqlAlchemy automatic labelling
-        try:
-            return table.c[column.column].label(column.dataframe_column_name())
-        except KeyError:
-            # If column.column is not in table.c it may be because the column names are case
-            # insensitive. If so, the schema can be in upper case (what oracle considers as
-            # case insensitive) but the keys in table.c are in lower case (what sqlalchemy
-            # considers as case insensitive).
-            try:
-                return table.c[column.column.lower()].label(column.dataframe_column_name())
-            except KeyError:
-                raise ImproperMappingError(
-                    f"Column '{column.column}' not found in table '{column.table}'."
-                )
-
-    def get_table(self, column: SqlColumn) -> Table:
-        """ Get the sql alchemy table corresponding to the SqlColumn (custom type)
-        from the analysis.
-        """
-        return Table(
-            column.table, self.metadata, schema=column.owner, keep_existing=True, autoload=True,
-        )
-
     @staticmethod
     @Timer("time_extractor_split", "time to split dataframe")
     def split_dataframe(df: Query, analysis: Analysis):
+        # TODO maybe this could be replaced by a group_by?
```

Comment:
Good idea ! We already discussed the fact that the current implem is dangerous if we paginate the results of the sql query right ? could you create an issue for this ?

Reply:
oh you're way ahead of me... #162